### ISSUE TITLE
Generators.asyncInput?

### ISSUE THREAD

**Issue by mbostock** [observablehq/stdlib#126]
It’d be nice to have an async iterator version of Generators.input.

Same also with Generators.queue and Generators.observe?

**Comment by tomlarkworthy** [comment]
Yes, I ran into this during https://talk.observablehq.com/t/metaprogramming-with-peek-peekfirst/4540

The runtime emits a stream of values and which error is not know in advance. This is best modelled as an async generator of concrete values, rather than a generator of promises. An async generator prevents the caller from calling next() until the value is known.
